src/three-idx.ts

```diff
@@ -74,40 +85,52 @@ export class ThreeIDX {
     return this._v03ID || `did:3:${this.docs.threeId.id.baseID.toString()}`
   }
 
-  async create3idDoc(docParams: Record<string, any>): Promise<void> {
-    this.docs.threeId = await this.ceramic.createDocument('tile', docParams, {
-      anchor: false,
-      publish: false,
-    })
+  async create3idDoc(docParams: ThreeIdState): Promise<void> {
+    this.docs.threeId = await TileDocument.create(
+      this.ceramic,
+      docParams.content,
+      docParams.metadata,
+      {
+        anchor: false,
+        publish: false,
+      }
+    )
   }
 
   get3idVersion(): string {
-    const docId = this.docs.threeId.anchorCommitIds.pop()
+    const anchorCommitIds = this.docs.threeId.anchorCommitIds
+    const docId = anchorCommitIds[anchorCommitIds.length - 1]
     return docId ? docId.commit.toString() : '0'
   }
 
-  async loadDoc(name: string, controller: string, family: string): Promise<void> {
-    this.docs[name] = await this.ceramic.createDocument(
-      'tile',
-      { deterministic: true, metadata: { controllers: [controller], family } },
+  async loadDoc(name: string, controller: string, family: string): Promise<TileDocument> {
+    const stream = await TileDocument.create<Record<string, any>>(
+      this.ceramic,
+      null,
+      { controllers: [controller], family: family, deterministic: true },
       { anchor: false, publish: false }
     )
+    this.docs[name] = stream
+    return stream
   }
 
   async createAuthLinkUpdate({ did }: NewAuthEntry): Promise<AuthLinkDocUpdate> {
     const didString = did.id
-    await this.loadDoc(didString, didString, 'authLink')
-    await this.ceramic.pin.add(this.docs[didString].id)
+    const tile = await this.loadDoc(didString, didString, 'authLink')
+    await this.ceramic.pin.add(tile.id)
+    const commit = await tile.makeCommit({ did }, { did: this.id })
     return {
-      commit: await TileDoctype._makeCommit(this.docs[didString], did, { did: this.id }),
-      docid: this.docs[didString].id,
+      commit: commit,
+      docid: tile.id,
       did: didString,
     }
   }
 
   async applyAuthLinkUpdate({ docid, commit, did }: AuthLinkDocUpdate): Promise<void> {
+    // @ts-ignore
     if (this.docs[did].content !== this.id) {
-      await this.ceramic.applyRecord(docid, commit)
+      await this.ceramic.applyCommit(docid, commit)
+      await this.docs[did].sync()
```

Comment:
What does the sync() method call achieve here?

Reply:
It synchronises state of `this.docs[did]` to what is inside Ceramic. Ceramic instance knows the state, but the doc does not after `applyCommit`.

Reply:
Why is that needed? It seems to me that that state should be the same as the state before the call to `sync()` if anything, we should make this call just before we call `applyCommit`?

Reply:
If no `sync`, `this.docs[did].content` would be same as before `applyCommit`, which is not what is expected. The applied commit should change the state. Really, I see no reason why this `applyCommit` within `applyAuthLinkUpdate` should not be merged with `createAuthLinkUpdate` into vanilla `this.docs[did].update`. That is not possible though due to that did switching we discussed on Discord.

Reply:
Anyway, there are a couple more of `sync` calls here and there. Better treat them as `HERE BE DRAGONS` sign to check when subscriptions are in place.

Reply:
Correct me if I'm wrong, but doesn't the `applyCommit` method return an instance of the Stream? If so we could just write this to `this.docs[did]` here.